azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java

```diff
@@ -111,9 +92,7 @@ public AzkabanServer getApplication() {
 
   @Override
   public void init(final ServletConfig config) throws ServletException {
-    this.application =
-        (AzkabanServer) config.getServletContext().getAttribute(
-            AzkabanServletContextListener.AZKABAN_SERVLET_CONTEXT_KEY);
+    this.application = SERVICE_PROVIDER.getInstance(AzkabanWebServer.class);
```

Comment:
I don't suspect that the class will be **injected** here. This should be a fetch since Guice maintains only a single instance of the web server class. Although, this may fail spectacularly if the injection happens while constructing the web server itself in which case this should throw up some sort of a circular dependency error. None of that seems to be happening though. Else the build should fail because of injection tests in `AzkabanWebServerTest`